portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -23,10 +25,21 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
-
-    Key commentEntityKey = KeyFactory.createKey("Comment", id);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(commentEntityKey);
+    UserService userService = UserServiceFactory.getUserService();
+    String commentEmail = request.getParameter("email");
+    String userEmail = request.getParameter("user");
+    //if user did not make the comment or if user is not logged in 
+    if (!commentEmail.equals(userEmail) || !userService.isUserLoggedIn()){
```

Comment:
This is excellent, you're doing server-side validation to ensure that nobody is pretending to be somebody else or hit this endpoint independently. This is true use of User Authentication. Nice work!